source/EffectfulForcing/MFPSAndVariations/ContinuityProperties.lagda

```diff
@@ -0,0 +1,913 @@
+member-implies-below-max : (s : List ℕ) (i : ℕ) → member i s → i ≤ℕ maximum s
+member-implies-below-max (m ∷ ns) m in-head     = max-≤-upper-bound m (maximum ns)
+member-implies-below-max (n ∷ ns) m (in-tail p) =
+ ≤-trans m _ _ IH (max-≤-upper-bound' (maximum ns) n)
+  where
+   IH =(member-implies-below-max ns m p)
+
+＝⦅⦆-implies-＝⟪⟫ : {X : 𝓤₀  ̇} (α α′ : ℕ → X) (s : List ℕ)
+                  → α ＝⦅ succ (maximum s) ⦆ α′
+                  → α ＝⟪ s ⟫ α′
+＝⦅⦆-implies-＝⟪⟫ α α′ s t = ＝⟪⟫₀-implies-＝⟪⟫ α α′ s †
+ where
+  m : ℕ
+  m = succ (maximum s)
+
+  † : α ＝⟪ s ⟫₀ α′
+  † i p = agreement→ α α′ m t i (member-implies-below-max s i p)
+
+continuity-implies-continuity₀ : (f : Baire → ℕ)
+                               → is-continuous f → is-continuous₀ f
+continuity-implies-continuity₀ f c = †
+ where
+  † : is-continuous₀ f
+  † α = m , γ
+   where
+    s = pr₁ (c α)
+    m = succ (maximum s)
+
+    γ : (α′ : Baire) → α ＝⦅ m ⦆ α′ → f α ＝ f α′
+    γ α′ p = pr₂ (c α) α′ (＝⦅⦆-implies-＝⟪⟫ α α′ s p)
+
+\end{code}
+
+\section{`is-continuous₀` implies `is-continuous`}
+
+We now address the converse direction which is harder.
+
+We first define a `range` function such that `range n` is the list `[0..n]` and
+prove its completeness.
+
+\begin{code}
+
+range : ℕ → List ℕ
+range zero     = [ 0 ]
+range (succ n) = succ n ∷ range n
+
+range-succ : (i n : ℕ) → member i (range n) → member (succ i) (range (succ n))
+range-succ zero     zero     p            = in-head
+range-succ (succ i) zero     (in-tail ())
+range-succ zero     (succ n) (in-tail p)  = in-tail (range-succ zero n p)
+range-succ (succ i) (succ i) in-head      = in-head
+range-succ (succ i) (succ n) (in-tail p)  = in-tail (range-succ (succ i) n p)
+
+range-is-complete : (i n : ℕ) → i ≤ℕ n → member i (range n)
+range-is-complete zero     zero     p = in-head
+range-is-complete zero     (succ n) p = in-tail (range-is-complete zero n p)
+range-is-complete (succ i) (succ n) p = range-succ i n (range-is-complete i n p)
+
+\end{code}
+
+We combine all these into a final lemma that we need:
+
+\begin{code}
+
+＝⟪⟫-range-implies-＝⦅⦆ : (α α′ : Baire) (n : ℕ)
+                        → α ＝⟪ range n ⟫ α′
+                        → α ＝⦅ n ⦆ α′
+＝⟪⟫-range-implies-＝⦅⦆ α α′ n p = agreement← α α′ n †
+ where
+  † : (j : ℕ) → j <ℕ n → α j ＝ α′ j
+  † j q = ＝⟪⟫-implies-＝⟪⟫₀ α α′ (range n) p j ‡
+   where
+    ‡ : member j (range n)
+    ‡ = range-is-complete j n (<-coarser-than-≤ j n q)
+
+\end{code}
+
+The result we want now follows easily:
+
+\begin{code}
+
+continuity₀-implies-continuity : (f : Baire → ℕ)
+                               → is-continuous₀ f → is-continuous f
+continuity₀-implies-continuity f c α = range m , γ
+ where
+  m = modulus-at₀ f c α
+
+  γ : (α′ : Baire) → α ＝⟪ range m ⟫ α′ → f α ＝ f α′
+  γ α′ p = pr₂ (c α) α′ (＝⟪⟫-range-implies-＝⦅⦆ α α′ m p)
+
+\end{code}
+
+Finally, we record the logical equivalence as a fact in itself.
+
+\begin{code}
+
+continuity₀-iff-continuity : (f : Baire → ℕ)
```

Comment:
I would say `continuous₀-iff-continuous` or `continuity₀-equiv-to-continuity` for grammatical reasons.

Reply:
Suggestion implemented.